program: add permision.js,validation.js(39517)

utkarsh-successive opened the pull request.

utkarsh-successive committed:
program: add permision.js,validation.js(39517)

utkarsh-successive committed:
program: add diamond.js,equlitral.js(39517)

rahulsainidev commented on extra/utils/validation.js:
```diff
@@ -0,0 +1,55 @@
+
+function validateUsers(users)
+{
+    let validUsers=0;
```
Declaring variables for valid and invalid count is unnecessary you can take advantage of .length property of array to get the no of record in array. 
For example to get valid user count you can simply write `userValid.length`

rahulsainidev reviewed (changes requested):
In permision.js you have not handled the "all" permission case if you call `console.log(hasPermission('getUsers','head-trainer','write'));` it will give false which is not correct.


IN

utkarsh-successive committed:
Update permision.js

utkarsh-successive committed:
Update permision.js

rahulsainidev commented on extra/utils/permision.js:
```diff
@@ -0,0 +1,70 @@
+        if(permissions[moduleName][permissionType]!==undefined)
+        {
+            console.log("Permission Type ="+permissionType)
+            if(permissions[moduleName][permissionType]!=='all')
```
You are tring to compare array with string value. There is no need to do that since you have already checked for permissions[moduleName][permissionType]!==undefined

rahulsainidev commented on extra/utils/permision.js:
```diff
@@ -0,0 +1,70 @@
+                    return true;
+                }
+            }
+            if(permissions[moduleName][permissionType].length>0)
```
No need to check length if array is empty indexOf will return -1. So no need for this check.

utkarsh-successive committed:
Update permision.js

sir i am changing this code to new logic . Because old logic use more if and else.
